src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    // Constants for setting the size and increasing the size of the list by default.
```

Comment:
You don't need to comment the code, 
the code should be easy to read without comments